lib/vrl/stdlib/src/parse_duration.rs

```diff
@@ -117,96 +117,85 @@ impl Expression for ParseDurationFn {
-    //         ),
-    //         (
-    //             map![],
-    //             Ok(1.005.into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1005ms")), "s"),
-    //         ),
-    //         (
-    //             map![],
-    //             Ok(0.0001.into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("100ns")), "ms"),
-    //         ),
-    //         (
-    //             map![],
-    //             Ok(86400.0.into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1d")), "s"),
-    //         ),
-    //         (
-    //             map![],
-    //             Ok(1000000000.0.into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1 s")), "ns"),
-    //         ),
-    //         (
-    //             map![],
-    //             Err("function call error: unable to parse duration: 'foo'".into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("foo")), "µs"),
-    //         ),
-    //         (
-    //             map![],
-    //             Err("function call error: unable to parse duration: '1'".into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1")), "ns"),
-    //         ),
-    //         (
-    //             map![],
-    //             Err("function call error: unknown duration unit: 'w'".into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1w")), "ns"),
-    //         ),
-    //         (
-    //             map![],
-    //             Err("function call error: unknown unit format: 'w'".into()),
-    //             ParseDurationFn::new(Box::new(Literal::from("1s")), "w"),
-    //         ),
-    //     ];
-
-    //     let mut state = state::Program::default();
-
-    //     for (object, exp, func) in cases {
-    //         let mut object: Value = object.into();
-    //         let got = func
-    //             .resolve(&mut ctx)
-    //             .map_err(|e| format!("{:#}", anyhow::anyhow!(e)));
-
-    //         assert_eq!(got, exp);
-    //     }
-    // }
+    use super::*;
+
+    test_function![
+        parse_duration => ParseDuration;
+
+        one {
+            args: func_args![value: "30s",
+                             unit: "m"],
+            want: Ok(value!(0.5)),
+            tdef: TypeDef::new().fallible().float(),
+        }
+
+        two {
+            args: func_args![value: "1200ms",
+                             unit: "s"],
+            want: Ok(value!(1.2)),
+            tdef: TypeDef::new().fallible().float(),
+        }
+
+        three {
+            args: func_args![value: "100ms",
+                             unit: "ms"],
+            want: Ok(100.0),
+            tdef: TypeDef::new().fallible().float(),
+        }
+
+       four {
+           args: func_args![value: "1005ms",
+                            unit: "s"],
+           want: Ok(1.005),
+           tdef: TypeDef::new().fallible().float(),
+       }
+        five {
```

Comment:
There's some weird indentation going on here (three is indented one more than four, and five is missing a newline).

Also, the naming isn't particularly useful, but I guess I guess we can leave that as is.